### PR TITLE
Mat 8605 allow creating new test cases on versioned measure

### PR DESCRIPTION
**Pull request by sb-prateekkeerthi**
## MADiE PR

Jira Ticket: [MAT-8605](https://jira.cms.gov/browse/MAT-8605)
(Optional) Related Tickets:

### Summary
allow creating new test cases on versioned measure

### All Submissions
* [ ] This PR has the JIRA linked.
* [ ] Required tests are included.
* [ ] No extemporaneous files are included (i.e Complied files or testing results).
* [ ] This PR is merging into the **correct branch**.
* [ ] All Documentation needed for this PR is Complete (or noted in a TODO or other Ticket).
* [ ] Any breaking changes or failing automations are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package).
* [ ] All CDN/Web dependencies are hosted internally (i.e MADiE-Root Repo).

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose.
*  The tests appropriately test the new code, including edge cases.
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads.

### CODE REVIEW

File: src/util/useCheckCanEdit.ts [hunk 1]
```diff
@@ -4,12 +4,16 @@ import { Acl } from "@madie/madie-models/dist/Measure";
 const useCheckUserCanEdit = (
   createdBy: string,
   acls: Array<Acl>,
-  draft: boolean = true
+  draft: boolean = true,
+  EditTestsOnVersionedMeasures?: boolean
```
Comment (adongare) [inline src/util/useCheckCanEdit.ts]: camelCase would be consistent with other parameters

Comment (sb-prateekkeerthi) [reply src/util/useCheckCanEdit.ts]: fixed it

File: src/util/useCheckCanEdit.ts [hunk 2]
```diff
@@ -4,12 +4,16 @@ import { Acl } from "@madie/madie-models/dist/Measure";
...
   acls: Array<Acl>,
-  draft: boolean = true
+  draft: boolean = true,
+  editTestsOnVersionedMeasures?: boolean
 ): boolean => {
   const { getUserName } = useOktaTokens();
   const userName = getUserName();
   // versioned measures/libraries are always uneditable.
   if (!draft) {
+    if (editTestsOnVersionedMeasures) {
```
Comment (adongare) [inline src/util/useCheckCanEdit.ts]: can I edit if I'm not a owner/sharedwith user?

File: src/util/useCheckCanEdit.ts [hunk 3]
```diff
@@ -4,12 +4,16 @@ import { Acl } from "@madie/madie-models/dist/Measure";
...
+      return true;
```
Comment (jkotanchik-SB) [inline src/util/useCheckCanEdit.ts]: Needs to consider ownership/shared permissions. Story ACs call out "user with edit access":
> User with edit access to the measure can add a new test case to a versioned measures
User with edit access to the measure can clone existing test cases on a versioned measures
User with edit access to the measure can import test cases to a versioned measure (import as new test cases)

File: src/util/useCheckCanEdit.ts (hunk 2 above)
Comment (sb-prateekkeerthi) [reply src/util/useCheckCanEdit.ts]: thank you, I missed that scenario I have fixed it now

File: src/util/useCheckCanEdit.ts (hunk 3 above)
Comment (sb-prateekkeerthi) [reply src/util/useCheckCanEdit.ts]: thank you, I missed that scenario I have fixed it now

**Review by adongare — APPROVED** [review]

**Review by jkotanchik-SB — APPROVED** [review]
